packages/flutter/lib/src/cupertino/activity_indicator.dart

```diff
@@ -194,3 +205,119 @@ class _CupertinoActivityIndicatorPainter extends CustomPainter {
+///  * [CupertinoActivityIndicator], which is an iOS-style activity indicator that spins clockwise.
+///  * <https://developer.apple.com/design/human-interface-guidelines/progress-indicators/>
+class CupertinoLinearActivityIndicator extends _CupertinoProgressIndicator {
+  /// Creates a linear iOS-style activity indicator.
+  const CupertinoLinearActivityIndicator({
+    super.key,
+    required super.progress,
+    this.height = 4.5,
+    super.color,
+    super.semanticsLabel,
+    super.semanticsValue,
+  }) : assert(height > 0),
+       assert(progress >= 0.0 && progress <= 1.0);
+
+  /// The height of the line used to draw the linear activity indicator.
+  ///
+  /// Defaults to 4.5 units. Must be positive.
+  final double height;
+
+  @override
+  Color getValueColor(BuildContext context, {Color? defaultColor}) {
+    return color ?? CupertinoColors.activeBlue;
+  }
+
+  @override
+  State<CupertinoLinearActivityIndicator> createState() => _CupertinoLinearActivityIndicatorState();
+}
+
+class _CupertinoLinearActivityIndicatorState extends State<CupertinoLinearActivityIndicator>
+    with
+        SingleTickerProviderStateMixin<CupertinoLinearActivityIndicator>,
+        ProgressIndicatorMixin<CupertinoLinearActivityIndicator> {
+  @override
+  Widget build(BuildContext context) {
+    return buildSemanticsWrapper(
+      context: context,
+      child: ConstrainedBox(
+        constraints: BoxConstraints(minHeight: widget.height, minWidth: double.infinity),
+        child: CustomPaint(
+          painter: _CupertinoLinearActivityIndicator(
+            progress: widget.value,
+            color: widget.getValueColor(context),
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  bool get animating => false;
+
+  @override
+  Duration get duration => Duration.zero;
+}
+
+class _CupertinoLinearActivityIndicator extends CustomPainter {
+  _CupertinoLinearActivityIndicator({required this.progress, required this.color})
+    : _backgroundPaint = Paint()
+        ..color = CupertinoColors.systemFill
+        ..style = PaintingStyle.fill,
+      _progressPaint = Paint()
+        ..color = color
+        ..style = PaintingStyle.fill;
+
+  final double progress;
+
+  final Color color;
+
+  /// The background paint used to draw the full width of the progress bar.
+  ///
+  /// This paint object is created once and reused to fill the background
+  /// with a system fill color.
+  final Paint _backgroundPaint;
+
+  /// The paint used to draw the progress portion of the progress bar.
+  ///
+  /// This paint object is created once and reused to fill the progress area.
+  final Paint _progressPaint;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    // Draw the background of the progress bar.
+    canvas.drawRRect(
+      BorderRadius.all(Radius.circular(size.height / 2)).toRRect(Offset.zero & size),
+      _backgroundPaint,
+    );
+
+    // Draw the progress portion of the bar.
+    if (progress > 0) {
+      canvas.drawRRect(
```

Comment:
We should use RSuperellipse instead of RRect in cupertino widgets. For example, 

```suggestion
      canvas.drawRSuperellipse(
```

Among a few other places.

Reply:
We were talking about this in https://github.com/flutter/flutter/pull/170108#discussion_r2150889845

Visually, this seems closer to a stadium shape than a rounded superellipse to me, but I'm just going off the appearance. Did you come across any specs that clarify this? If so, where were they referenced?